chat-plugins/dice-commands.js

```diff
@@ -1,185 +1,168 @@
-/* Dice commands chat-plugin
- * by jd
-*/
-var tax = 0.10; // 10%
+/*Dice Plugin by SilverTactic (Silveee)*/
 
-function handleWinnings (bet) {
-	return bet - Math.round(bet * tax);
+'use strict';
+
+if (!global.dicegames) {
+	global.dicegames = {};
+	global.dicegames.count = 0;
+}
+let dicegames = global.dicegames;
+
+const INACTIVE_END_TIME = 60 * 1000; //1 minute
+
+function diceImg(num) {
+	switch (num) {
+		case 0:
+			return "http://i.imgur.com/nUbpLTD.png";
+		case 1:
+			return "http://i.imgur.com/BSt9nfV.png";
+		case 2:
+			return "http://i.imgur.com/eTQMVhY.png";
+		case 3:
+			return "http://i.imgur.com/3Y2hCAJ.png";
+		case 4:
+			return "http://i.imgur.com/KP3Za7O.png";
+		case 5:
+			return "http://i.imgur.com/lvi2ZZe.png";
+	}
+}
+
+class Dice {
+	constructor(room, amount, starter) {
+		this.room = room;
+		this.reward = amount;
+		this.players = [];
+		this.timer = setTimeout(() => {
+			this.room.add('|uhtmlchange|' + dicegames.count + '|<div class = "infobox">(This game of dice has been ended due to inactivity.)</div>').update();
+			delete dicegames[this.room.id];
+		}, INACTIVE_END_TIME);
+
+		let buck = (this.reward === 1 ? 'buck' : 'bucks');
+		this.startMessage = '<div class = "infobox"><b style = "font-size: 14pt; color: #24678d"><center><span style = "color: ' + Gold.hashColor(starter) + '">' + Tools.escapeHTML(starter) + '</span> has started a game of dice for <span style = "color: green">' + amount + '</span> ' + buck + '!</center></b><br>' +
+			'<center><img style = "margin-right: 15px;" src = "http://i.imgur.com/eywnpqX.png" width = "110" height = "110">' +
+			'<img style = "transform:rotateY(180deg); margin-left: 15px;" src = "http://i.imgur.com/eywnpqX.png" width = "110" height = "110"><br>' +
+			'<button name = "send" value = "/joindice" style = "margin: 1px;">Click to join!</button></center>';
+		this.room.add('|uhtml|' + (++dicegames.count) + '|' + this.startMessage + '</div>');
```

Comment:
There's no need to. Here, <code>room.add()</code> isn't being executed inside a callback statement, so the room'll automatically be updated